src/libraries/System.Reflection.MetadataLoadContext/src/System/Reflection/Runtime/BindingFlagSupport/PropertyPolicies.cs

```diff
@@ -83,5 +83,35 @@ public sealed override bool OkToIgnoreAmbiguity(PropertyInfo m1, PropertyInfo m2
 
         private static MethodInfo? GetAccessorMethod(PropertyInfo property) =>
             property.GetMethod ?? property.SetMethod;
+
+        private static MethodInfo? GetMostAccessibleAccessor(PropertyInfo property)
+        {
+            MethodInfo? getter = property.GetMethod;
+            MethodInfo? setter = property.SetMethod;
+
+            if (getter == null)
+                return setter;
+            if (setter == null)
+                return getter;
+
+            // Define accessibility ranking
+            static int GetAccessibilityRank(MethodInfo method)
+            {
+                MethodAttributes access = method.Attributes & MethodAttributes.MemberAccessMask;
+                return access switch
+                {
+                    MethodAttributes.Public => 4,
+                    MethodAttributes.Family => 3,          // protected
+                    MethodAttributes.Assembly => 3,        // internal
+                    MethodAttributes.FamORAssem => 3,      // protected internal
+                    MethodAttributes.FamANDAssem => 2,     // protected and internal
+                    MethodAttributes.Private => 1,
+                    _ => 0
+                };
+            }
+
+            // Return the setter if it's more accessible, otherwise return the getter (preserving original behavior when equal)
+            return GetAccessibilityRank(setter) > GetAccessibilityRank(getter) ? setter : getter;
```

Comment:
```suggestion
            // Return the setter if it's more accessible, otherwise return the getter (preserving original behavior when equal)
            return GetAccessibilityRank(setter) > GetAccessibilityRank(getter) ? setter : getter;

            // Define accessibility ranking
            static int GetAccessibilityRank(MethodInfo method)
            {
                MethodAttributes access = method.Attributes & MethodAttributes.MemberAccessMask;
                return access switch
                {
                    MethodAttributes.Public => 4,
                    MethodAttributes.Family => 3,          // protected
                    MethodAttributes.Assembly => 3,        // internal
                    MethodAttributes.FamORAssem => 3,      // protected internal
                    MethodAttributes.FamANDAssem => 2,     // protected and internal
                    MethodAttributes.Private => 1,
                    _ => 0
                };
            }
```

I think the generally agreed upon pattern has been that local functions are placed at the end of the method.

Reply:
I couldn't find that in https://github.com/dotnet/runtime/blob/664ac25dcc16cb1531f90765295ececf5b45a0e3/docs/coding-guidelines/coding-style.md.  If that truly is the guideline we should add it so that copilot can do a better job.

I'm fine to move this, however I personally didn't have a problem with it being inline.